docs/reference/targets.md

```diff
@@ -88,6 +88,29 @@ if (this.hasResultsTarget) {
 }
 ```
 
+## Addition and Removal Callbacks
+
+Target _element callbacks_ let you respond whenever a target element is added or
+removed within the controller's element.
+
+Define a method `[name]TargetAdded` or `[name]TargetRemoved` in the controller, where `[name]` is the name of the target you want to observe for additions or removals. The method receives the element as the first argument.
+
+Stimulus invokes each change callback any time its target elements are added or removed.
+
+```js
+export default class extends Controller {
+  static targets = [ "input" ]
+
+  inputTargetAdded(element) {
```

Comment:
@sstephenson to match the Controller's `connect()` and `disconnect()`, would it be better to name these `inputTargetConnect()` and `inputTargetDisconnect()`?

Reply:
My $0.02 is that `added`/`removed` makes more sense because you're addressing elements, not controllers.

Reply:
I wonder if this documentation would be better suited for the Lifecycle Callbacks page instead? And we can briefly mention it with a link on the Targets page.